bin/anvi-analyze-synteny

```diff
@@ -0,0 +1,102 @@
+                                     --annotation-source COG_FUNCTION \
+                                     --window-range 2:3 \
+                                     -o $output_dir/synteny_output_with_unknowns.tsv \
+                                     --analyze-unknown-functions
+
+            Unknown-function-mode test that records ngrams with unknown functions:\
+
+            >>> anvi-analyze-synteny -e $output_dir/external-genomes-cps.txt \
+                                     --annotation-source COG_FUNCTION \
+                                     --window-range 2:3 \
+                                     -o $output_dir/tsv.txt \
+                                     --analyze-unknown-functions
+"""
+
+import sys
+import argparse
+
+import anvio
+import anvio.terminal as terminal
+
+from anvio.synteny import NGram
+from anvio.errors import ConfigError, FilesNPathsError
+
+
+__author__ = "Developers of anvi'o (see AUTHORS.txt)"
+__copyright__ = "Copyleft 2015-2018, the Meren Lab (http://merenlab.org/)"
+__credits__ = []
+__license__ = "GPL 3.0"
+__version__ = anvio.__version__
+__maintainer__ = "A. Murat Eren"
+__email__ = "a.murat.eren@gmail.com"
+__requires__ = ['contigs-db']
+__provides__ = ['locus-fasta']
+__description__ = ("This program helps you extract synteny patterns from a group of similar loci or genomes. This "
+                   "is accomplished by deconstructing contigs into Ngrams (group of neighboring genes of which N is the number of genes) of gene "
+                   "functions using a sliding window method. Once completed, anvi'o will export a table with Ngrams counts "
+                   "for you to work with. By default anvi'o will ignore Ngrams that contain genes without annotations. If "
+                   "you would like to override this, you can use the --analyze-unknown-functions flag.")
+
+
+
+run = terminal.Run()
+progress = terminal.Progress()
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description=__description__)
+
+    # Essential input
+    groupA = parser.add_argument_group('Essential INPUT')
+    groupA.add_argument(*anvio.A('external-genomes'), **anvio.K('external-genomes', {"required": True}))
+    groupA.add_argument(*anvio.A('annotation-source'), **anvio.K('annotation-source', {"required": True}))
+    groupA.add_argument(*anvio.A('window-range'), **anvio.K('window-range', {"required": True}))
+    groupA.add_argument(*anvio.A('output-file'), **anvio.K('output-file', {"required": True})) 
+
+    # Additional input
+    groupB = parser.add_argument_group('Additonal INPUT')
+    groupB.add_argument('--analyze-unknown-functions',action="store_true",
+                        help="Provide this flag if you want anvi-analyze-synteny to take into account "
+                              "Ngrams that contain gene calls that have no annotation.")
+
+    args = anvio.get_args(parser)
+
+    try:
+        ngram = NGram(args)
+        ngram.report_ngrams_to_user()
```

Comment:
Why not just `ngram.report()`?

Reply:
Are you referring to renaming the method `report_ngrams_to_user` to `report`?